crates/worker/src/partition/shuffle.rs

```diff
@@ -280,24 +260,31 @@ where
         let state_machine = StateMachine::new(
             metadata,
             outbox_reader,
-            |msg| network_tx.send(msg),
+            |msg| async move {
+                let_assert!(Destination::Processor { partition_key, .. } = &msg.header.dest, "shuffle can only send messages to a partition processor");
+
+                let partition_id = restate_core::metadata()
+                    .partition_table()
+                    .find_partition_id(*partition_key)?;
+                let log_id = LogId::from(partition_id);
+
+                let payload = Payload::from(msg.encode_with_bincode()?);
+                bifrost().append(log_id, payload).await?;
```

Comment:
After an offline discussion, we concluded that we cannot let the process fail because of a single component because the process might run other services that are still properly working. Instead, the idea is to retry (indefinitely) and surface a potentially unhealthy state to the cluster controller. The cluster controller can then decide to shut down a certain component. This also allows to make global decisions in case there is a cluster-wide problem. 

I will add the logic to record a health status which is made available to the node services as a follow-up PR.

Reply:
Follow-up issue: https://github.com/restatedev/restate/issues/1266